scripts/cmake/ExternalProjectBoost.cmake

```diff
@@ -1,116 +1,39 @@
 include(ThirdPartyLibVersions)
 include(ExternalProject)
-# Set Boost version and which libraries to compile
-set(BOOST_LIBS_TO_BUILD
-	# chrono
-	# context
-	date_time
-	# exception
-	filesystem
-	# graph
-	# graph_parallel
-	# iostreams
-	# locale
-	# math
-	# mpi
-	# program_options
-	# python
-	# random
-	# regex
-	# serialization
-	# signals
-	system
-	# test
-	# thread
-	# timer
-	# wave
-)
 
 if(Boost_FOUND)
 	return()
 endif()
 
-# Set boost toolset
-if(MSVC11)
-	set(BOOST_TOOLSET msvc-11.0)
-elseif(MSVC12)
-	set(BOOST_TOOLSET msvc-12.0)
-elseif(MSVC14)
-	set(BOOST_TOOLSET msvc-14.0)
-elseif(APPLE)
-	set(BOOST_TOOLSET darwin)
-elseif(COMPILER_IS_CLANG)
-	set(BOOST_TOOLSET clang)
-elseif(COMPILER_IS_INTEL)
-	# Extracts first two version numbers, e.g. 13.0 from 13.0.0.20121010
-	string(REGEX MATCH ^[0-9]*.[0-9] INTEL_COMPILER_VERSION ${CMAKE_CXX_COMPILER_VERSION})
-	set(BOOST_TOOLSET intel-${INTEL_COMPILER_VERSION})
-elseif(COMPILER_IS_GCC)
-	set(BOOST_TOOLSET gcc)
-endif()
-
 # First check for system boost
 if(NOT Boost_INCLUDE_DIRS)
 	if(APPLE)
 		set(BOOST_ROOT $ENV{HOMEBREW_ROOT})
 	endif()
-	if(MSVC)
-		set(Boost_USE_STATIC_LIBS ON)
-		if(NOT DEFINED BOOST_LIBRARYDIR)
-			set(BOOST_LIBRARYDIR "$ENV{CMAKE_LIBRARY_SEARCH_PATH}/boost/lib${BITS}-${BOOST_TOOLSET};C:/boost/lib${BITS}-${BOOST_TOOLSET};$ENV{BOOST_ROOT}/lib${BITS}-${BOOST_TOOLSET}")
-			set(BOOST_INCLUDEDIR "$ENV{CMAKE_LIBRARY_SEARCH_PATH}/boost;C:/boost;$ENV{BOOST_ROOT}")
-		endif()
-	endif()
 	if(WIN32 AND COMPILER_IS_GCC)
 		set(BOOST_INCLUDEDIR "$ENV{CMAKE_LIBRARY_SEARCH_PATH}/include/boost*")
 	endif()
 	if(OGS_LIB_BOOST STREQUAL "System")
-		find_package(Boost 1.46.0 COMPONENTS ${BOOST_LIBS_TO_BUILD} REQUIRED)
+		find_package(Boost 1.46.0 REQUIRED)
 	elseif(OGS_LIB_BOOST STREQUAL "Default")
-		find_package(Boost 1.46.0 COMPONENTS ${BOOST_LIBS_TO_BUILD})
+		find_package(Boost 1.46.0)
```

Comment:
I could update the required version in #912 once this PR is merged, since my PR really changes the requirement.